### ISSUE TITLE
IPv6 for Dynamic DNS

### ISSUE THREAD

**Issue by donpotat** [ovh/manager#3919]
**Is your feature request related to a problem? Please describe.**
I'd like the possibility to update my current IPv4 and IPv6 automatically with the Dynamic DNS feature.

**Describe the solution you'd like**
For those who have dual stack at home, I'd like the possibility so the API from DynDNS will pick up IPv6 and IPv4 and update the A and AAAA records simultaneously.

**Comment by trnsnt** [comment]
Hello,

DynHost (Dynamic DNS feature) is a legacy feature that we want to fully refactor. We take note of your need, and we will add support of ipv6 in DynHost in the refactored version. Unfortunately I do not have any ETA to give you for now.

Thanks,

Eric - DNS team

**Comment by antleblanc** [comment]
Thanks for the update @trnsnt!

**Comment by Tycorc** [comment]
Any news regarding the ETA?

**Comment by Tom94** [comment]
Hi there, I'm another user who'd love to have ipv6 dynamic dns.

**Comment by djey47** [comment]
Hello, no update?

**Comment by anthonygraignic** [comment]
Hi, any update ?
Would love to fully use IPv6, 24 years after the RFC release :)

**Comment by qupfer** [comment]
That worked for me:

In normal circumstances, your IPv6 address is your global public address, so you can relay on your "ip -a" (linux) or ipconfig (windows) command to get your IP locally. 
After hacking a few lines in bash/powershell using curl or anything similar, you can use the ovh api:
https://api.ovh.com/console/#/domain/zone/{zoneName}/record~GET  to get ID of your record
https://api.ovh.com/console/#/domain/zone/{zoneName}/record/{id}~GET to get info if current setting
and finally set new one with: https://api.ovh.com/console/#/domain/zone/{zoneName}/record/{id}~PUT

**Comment by Norde** [comment]
@qupfer So we can update DDNS using an ipv6 but still not update simultaneously ipv4 and v6 ( as ipv6 will replace the v4 entry and vice versa) ?

**Comment by qupfer** [comment]
> @qupfer So we can update DDNS using an ipv6 but still not update simultaneously ipv4 and v6 ( as ipv6 will replace the v4 entry and vice versa) ?

Yes and no. My method is not a real DDNS update, Its just change config via API.
With `result = client.get(f'/domain/zone/{zoneName}/record', fieldType='A',subDomain=subDomain)`
you could receive the record ID of the A-Recroderd and change it in the same way (`client.put(f'/domain/zone/{zoneName}/record/{recordID}',subDomain=subDomain,target=1.2.3.4,ttl=0)
`)
Extend the script, and you could update both. 


However, take the update in two different steps is not the worst idea.
At my home, I use different containers (similar to VMs) for different jobs. They are reachable through IPv4 and IPv6. 
The problem is: They have all the same IPv4 address (different portforwarding for IPv4 (NAT), but differen IPv6 adresses (no NAT on IPv6)).
A "single-update-Approach" would breake some connection in my setup, as the AAAA Record would only match one service. 
At the moment, I have an "univeral" ipv4 adress (updateded from the router as an old fashioned dnydns host)  and a servie-specific subdomain, updated from the host itself. It realy depends what you want to achieve and what your setup looks like ( a single host, a vm-host, multiplae host....)

**Comment by HDebeuf** [comment]
Any update on this feature request enabling both ipv4 and ipv6 support with DynHost?
Thank you.

**Comment by Norde** [comment]
> 

Thanks a lot for this update @qupfer!
(sorry for the late reply: I missed my notifications)

**Comment by PerspectivesLab** [comment]
still impossible to do ipv6 dyndns on ovh in 2024 ...

**Comment by server-home-KaKi87** [comment]
Hello,
Any news on this ?
Thanks

**Comment by trnsnt** [comment]
Hello,

IPv6 for DynDns will be supported in the coming weeks :)

Eric - DNS team

**Comment by gyptazy** [comment]
> IPv6 for DynDns will be supported in the coming weeks :)

Glad to head and almost unbelievable after all the years...

**Comment by PerspectivesLab** [comment]
do you know when this will be released ? its months now @trnsnt 
"Seule une IPv4 peut etre utilisée pour la mise en place d'un DynHost. Les IPv6 sont indisponibles."
from: https://help.ovhcloud.com/csm/fr-dns-dynhost?id=kb_article_view&sysparm_article=KB0051644

thx

**Comment by trnsnt** [comment]
Hello @PerspectivesLab 
It is ok using apiv6 ! 
This will be possible using manager when https://github.com/ovh/manager/pull/11307 will be prodded !

thx

**Comment by lippl** [comment]
Hi @trnsnt,
thanks for the implementation!
From what I tested so far, it is now possible to set an IPv6 address via manager web-ui and update url. However when setting the AAAA-Record, the A-Records is cleared, hence a dual-stack scenario is not possible.
The initial request and the reasonable way would be to have an A and AAAA record simulatneously. Hence it would also require two separate input fields in the gui and an additional myipv6 parameter for the request url.
Could you please look into this again?
Thanks :)

**Comment by PerspectivesLab** [comment]
Hi @lippl, 
i still cant see the option for ipv6 in the dynhost section of the ovh manager, how did you get there ?
thx

**Comment by lippl** [comment]
@PerspectivesLab there is no extra field or label. Just use the same field as for ipv4 before. Merely the input validation is allowing ipv6 format now.

**Comment by DerDreschner** [comment]
@trnsnt : I don't want to bother, but do you have any updates for us here? :) Thanks!!

**Comment by Neustradamus** [comment]
To follow this ticket.

**Comment by Kwasow** [comment]
Hey, I've set up DynDNS with an IPv6 address today and it seems to be working. I think this issue can be closed

**Comment by lippl** [comment]
Hey @Kwasow, am I right in assuming, that your DynDnS entry resolves to IPv6 only then? The issue is about being able to set and resolve IPv4 (A) AND IPv6 (AAAA) at the same time.

**Comment by Kwasow** [comment]
Oh, my bad then @lippl, I thought this was only about IPv6 not resolving at all. I don't have a public IPv4 address, so I can't help with that, unfortunately.

**Comment by antoinbo** [comment]
Hi everyone!

I already wanted to set an IPv6 address to a DynHost for many years, and finally seen it is now possible! But, I was also stuck to set both IPv4 and IPv6 address to the same DynHost.

However, I discovered that multiple DynHosts can be created for the same subdomain ([Any A or AAAA record with a TTL (Time To Live) of 60 seconds is considered DynHost.](https://help.ovhcloud.com/csm/en-gb-dns-dynhost?id=kb_article_view&sysparm_article=KB0051640)), so:
* `dynhost.example.net` targeting `0.0.0.0`
* `dynhost.example.net` targeting `::`

This creates two records in the DNS Zone, respectively A and AAAA.
Those can then be updated independently using one unique account (e.g., `example.net-dynhost`).

Update the A and AAAA records with two requests:
```
# curl -u "$LOGIN:$PASSWORD" https://dns.eu.ovhapis.com/nic/update?system=dyndns&hostname=$HOSTNAME&myip=$IP
curl -u "example.net-dynhost:$PASSWORD" https://dns.eu.ovhapis.com/nic/update?system=dyndns&hostname=dynhost.example.net&myip=127.0.0.1
curl -u "example.net-dynhost:$PASSWORD" https://dns.eu.ovhapis.com/nic/update?system=dyndns&hostname=dynhost.example.net&myip=::1
```

Lookup the subdomain returns the two versions of IP addresses 😃
```
Non-authoritative answer:
Name:   dynhost.example.net
Address: 127.0.0.1
Name:   dynhost.example.net
Address: ::1
```

Nota: Creating more than one A or AAAA record per subdomain using DynHost return `No record found` error.

**Comment by poikilotherm** [comment]
Thanks @antoinbo this was very helpful! Your suggestions still works in 2026, now using this to update via OpenWRT DDNS scripts at OVH.
